src/plugins/data_source_management/public/components/data_source_view/data_source_view.tsx

```diff
@@ -111,11 +116,22 @@ export class DataSourceView extends React.Component<DataSourceViewProps, DataSou
           this.props.onSelectedDataSources([{ id: optionId, label: selectedDataSource.title }]);
         }
       } catch (error) {
-        handleDataSourceFetchError(
-          this.onError.bind(this),
-          this.props.notifications!,
-          this.props.onSelectedDataSources
-        );
+        // return [] to customer when no default data source
+        if (!defaultDataSource) {
+          this.props.onSelectedDataSources?.([]);
+        }
+        this.setState({
+          defaultDataSource: this.getFilteredDefaultDataSource(defaultDataSource),
+        });
+        const handleDataSourceViewErrorParams = {
+          changeState: this.onError.bind(this),
+          notifications: this.props.notifications!,
+          failedDataSourceId: optionId,
+          defaultDataSource: this.state.defaultDataSource,
```

Comment:
so here we are setting the default data source at line 124, then has to wait until it is set then get the value? Can we extract the this.getFilteredDefaultDataSource(defaultDataSource) as const?

Reply:
Updated